lib/std/zig/parse.zig

```diff
@@ -3590,7 +3583,7 @@ const Parser = struct {
         _ = try p.parsePtrPayload();
 
         const then_expr = try bodyParseFn(p);
-        if (then_expr == 0) return p.fail(.invalid_token);
+        assert(then_expr != 0);
```

Comment:
how do you rule out that code can trip this assertion?

Reply:
I changed the calls to `parseIf` to use `expectExpr` which return `error.ParseError` if no node was found.